tests/org.eclipse.e4.ui.tests.css.core/src/org/eclipse/e4/ui/tests/css/core/parser/MediaRulesTest.java

```diff
@@ -29,18 +30,22 @@
 public class MediaRulesTest {
 
 	@Test
-	@Disabled("//THIS TEST KNOWN TO FAIL Dec 16/08")
 	void testMediaRule() throws Exception {
 		String css = """
-			@media screen, print {
-			BODY { line-height: 1.2 }
-			}
-			Label { background-color: #FF0000 }""";
+				@media screen, print {
+					BODY { line-height: 1.2 }
+					Label { background-color: #FFFFFF }
+				}
+				BODY { line-height: 1.3 }
+				Label { background-color: #FF0000 }""";
```

Comment:
You tested the same thing twice here: that an attribute outside the `@media` is not overwritten by those inside the `@media`.
You could test this only once and also test that the order in which the `@media` statements appear doesn't play any role _i.e._ use this...

```suggestion
@media screen, print {
	BODY { line-height: 1.2 }
	Label { background-color: #FFFFFF }
}
BODY { line-height: 1.3 }

@media screen, print {
	BODY { line-height: 1.4 }
}
""";
```

... and adapt the assertions (see below)

Reply:
You still need to change the content of `String css`